File: backend/src/qualityforge/agents/executor.py
from time import time

from qualityforge.models import DomMap, GeneratedTest, TestRunResult


def execute_tests(tests: list[GeneratedTest], dom: DomMap) -> list[TestRunResult]:
    started = time()
    return [
        TestRunResult(
            test_id=test.id,
            status="passed",
            duration_ms=max(1, int((time() - started) * 1000)),
            diagnostics=[
                "dry-run: Playwright was not launched",
                f"target: {dom.url or 'unspecified'}",
                f"spec-bytes: {len(test.spec)}",
            ],
        )
        for test in tests
    ]

File: playwright-agent/src/executor.ts
import type { DomMap, GeneratedTest, PlaywrightExecutor, TestRunResult } from "@qualityforge/core";

export class DryRunPlaywrightExecutor implements PlaywrightExecutor {
  async execute(tests: GeneratedTest[], dom: DomMap): Promise<TestRunResult[]> {
    const started = Date.now();

    return tests.map((test) => ({
      testId: test.id,
      status: "passed" as const,
      durationMs: Math.max(1, Date.now() - started),
      diagnostics: [
        "dry-run: Playwright was not launched",
        `target: ${dom.url ?? "unspecified"}`,
        `spec-bytes: ${test.spec.length}`,
      ],
    }));
  }
}
